pkg/types/options.go

```diff
@@ -97,14 +97,18 @@ type Options struct {
 	StoreResponse bool
```

Comment:
Overall, the code is well-structured and readable. However, there are a few areas where it could be improved:

1. **Error Handling**: Both `ParseCustomHeaders` and `ParseHeadlessOptionalArguments` methods are not handling the case when the header or argument is not in the correct format. It would be beneficial to add error handling or at least log a warning when this happens.

```go
// Existing code
if headerParts := strings.SplitN(v, ":", 2); len(headerParts) >= 2 {
    customHeaders[strings.Trim(headerParts[0], " ")] = strings.Trim(headerParts[1], " ")
}

// Suggested change
if headerParts := strings.SplitN(v, ":", 2); len(headerParts) >= 2 {
    customHeaders[strings.Trim(headerParts[0], " ")] = strings.Trim(headerParts[1], " ")
} else {
    log.Warnf("Invalid header format: %s", v)
}
```

2. **Code Duplication**: There's some code duplication in `ParseCustomHeaders` and `ParseHeadlessOptionalArguments`. Consider abstracting the common logic into a separate function.

```go
// Existing code
if argParts := strings.SplitN(v, "=", 2); len(argParts) >= 2 {
    key := strings.TrimSpace(argParts[0])
    value := strings.TrimSpace(argParts[1])
    if key != "" && value != "" {
        optionalArguments[key] = value
    }
}

// Suggested change
func parseKeyValuePair(pair string, separator string) (string, string) {
    parts := strings.SplitN(pair, separator, 2)
    if len(parts) < 2 {
        log.Warnf("Invalid key-value pair format: %s", pair)
        return "", ""
    }
    return strings.TrimSpace(parts[0]), strings.TrimSpace(parts[1])
}
```

3. **Commenting**: The comments on the struct fields are quite good, but there are none on the methods. Adding comments on what each method does would improve the code readability and maintainability.

```go
// ParseCustomHeaders parses the custom headers from the options
func (options *Options) ParseCustomHeaders() map[string]string {
    ...
}

// ParseHeadlessOptionalArguments parses the optional arguments for headless mode from the options
func (options *Options) ParseHeadlessOptionalArguments() map[string]string {
    ...
}
```

4. **Constant Strings**: Consider defining constants for strings that are used multiple times in the code like ":", "=" to avoid typos and make it easier to change in the future.

```go
const (
    HeaderSeparator = ":"
    ArgSeparator = "="
)
``